web/src/components/map.js

```diff
@@ -2,8 +2,22 @@ import mapboxgl from 'mapbox-gl';
 import 'mapbox-gl/dist/mapbox-gl.css';
 
 export default class Map {
-  _setupMapColor() {
-    if (this.map.isStyleLoaded() && this.map.getLayer('clickable-zones-fill') && this.co2color) {
+  _setupMapColor(theme) {
+    if (theme) {
+      console.log(theme.clickableFill);
+      if (this.map.isStyleLoaded() && this.map.getLayer('clickable-zones-fill') && this.co2color) {
+        console.log(theme.clickableFill);
+        const co2Range = theme.co2Scale.steps;
+        const stops = co2Range.map(d => [d, this.co2color(d)]);
+        this.map.setPaintProperty('clickable-zones-fill', 'fill-color', {
+          default: theme.clickableFill,
+          property: 'co2intensity',
+          stops,
+        });
+        this.map.setPaintProperty('zones-line', 'line-color', theme.strokeColor);
+        this.map.setPaintProperty('zones-line', 'line-width', theme.strokeWidth);
+      }
+    } else if (this.map.isStyleLoaded() && this.map.getLayer('clickable-zones-fill') && this.co2color) {
```

Comment:
why not keep this if/else as top level, and have the `if (theme)` inside instead? It will be much cleaner.